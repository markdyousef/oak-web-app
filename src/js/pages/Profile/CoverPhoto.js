import React, { PropTypes } from 'react';

import css from './Profile.css';

const CoverPhoto = ({ picture }) => {
    return (
        <div className={css.cover}>
            <img src={picture} alt="profile cover" />
        </div>
    );
};

CoverPhoto.propTypes = {
    picture: PropTypes.string
};

CoverPhoto.defaultProps = {
    picture: null
};

export default CoverPhoto;
