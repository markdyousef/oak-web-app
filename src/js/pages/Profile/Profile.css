:root {
    --primary-color: #5ed892;
    --main-black-color: #131517;
    --light-black-color: #2d3438;
    --dark-gray-color: #737a80;
    --main-gray-color: #f2f2f2;
    --light-gray-color: #fafafa;
    --main-white-color: #fff;
}

.container {
    width: 100%;
}

.cover {
    width: 100%;
    height: 150px;
    background-color: var(--dark-gray-color);
}

.nav {
    width: 100%;
    height: 50px;
    background-color: #fff;
    border-bottom: 1px solid var(--main-gray-color);
    display: flex;
    align-items: center;
}

.avatar {
    margin-left: 40px;
    height: 100px;
    width: 100px;
    border-radius: 100px;
    background-color: var(--main-white-color);
    border: 1px solid var(--main-gray-color);
}

.infoContainer {
    width: 100%;
    height: 200px;
    background-color: var(--main-white-color);
    border-bottom: 1px solid var(--main-gray-color);
    padding: 40px;
}

.name {
    display: flex;
    align-items: baseline;
}

.name h1 {
    color: var(--main-black-color);
    font-weight: bold;
    font-size: 18px;
    margin-right: 5px;
}

.name h2 {
    color: var(--dark-gray-color);
}
