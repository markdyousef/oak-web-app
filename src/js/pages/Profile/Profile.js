import React, { Component, PropTypes } from 'react';

import css from './Profile.css';

class Profile extends Component {
    static propTypes = {};
    constructor() {
        super();
        this.state = {}
    }
    render() {
        console.log('hello');
        return (
            <div className={css.container}>
                PROFILE
            </div>
        );
    }
}

export default Profile;
