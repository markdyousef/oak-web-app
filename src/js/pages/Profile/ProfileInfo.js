import React, { Component, PropTypes } from 'react';

import css from './Profile.css';

class ProfileInfo extends Component {
    static propTypes = {
        name: PropTypes.string.isRequired,
        username: PropTypes.string.isRequired,
        description: PropTypes.string
    }
    static defaultProps = {
        description: null
    }
    constructor() {
        super();
        this.state = {};
    }
    render() {
        const { name, username } = this.props;
        return (
            <div className={css.infoContainer}>
                <div className={css.name}>
                    <h1>{name}</h1>
                    <h2>{username}</h2>
                </div>
            </div>
        );
    }
}

export default ProfileInfo;
