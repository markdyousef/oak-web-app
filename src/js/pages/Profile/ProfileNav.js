// @flow
import React, { Component, PropTypes } from 'react';
import { Link } from 'react-router';

import css from './Profile.css';

class ProfileNav extends Component {
    static propTypes = {}
    constructor() {
        super();
        this.state = {}
    }
    render() {
        return (
            <div className={css.nav}>
                <div className={css.avatar} />
                <Link to="profile" activeClassName={css.activeTab}>Overview</Link>
                <Link activeClassName={css.activeTab}>Insights</Link>
            </div>
        );
    }
}

export default ProfileNav;
