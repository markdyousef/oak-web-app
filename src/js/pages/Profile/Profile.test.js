import React from 'react';
import renderer from 'react-test-renderer';
import Profile from './Profile';

it('renders correctly', () => {
    const component = renderer.create(
        <Profile />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
