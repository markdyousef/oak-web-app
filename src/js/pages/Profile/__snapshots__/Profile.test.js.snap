exports[`test renders correctly 1`] = `
<div
  className={undefined}>
  PROFILE
</div>
`;
